apps/src/templates/sectionProgress/progressTables/ProgressTableLevelBubble.jsx

```diff
@@ -68,6 +68,130 @@ const styles = {
   }
 };
 
+function levelStyle(props) {
+  const {levelStatus, levelKind, disabled} = props;
+  return {
+    ...(!disabled && progressStyles.hoverStyle),
+    ...progressStyles.levelProgressStyle(levelStatus, levelKind, disabled)
+  };
+}
+
+function mainStyle(props) {
+  return {
+    ...styles.main,
+    ...levelStyle(props)
+  };
+}
+
+function largeStyle(props) {
+  const {disabled, bonus} = props;
+  return {
+    ...mainStyle(props),
+    ...styles.large,
+    ...(disabled && bonus && styles.bonusDisabled)
+  };
+}
+
+function largeContentStyle(props) {
+  const {bonus, paired} = props;
+  return {
+    ...progressStyles.flex,
+    ...styles.contents,
+    fontSize: paired || bonus ? 14 : 16
+  };
+}
+
+function UnpluggedBubble(props) {
+  return (
+    <div
+      style={{
+        ...styles.main,
+        ...styles.unplugged,
+        ...levelStyle(props)
+      }}
+    >
+      <Content {...props} />
+    </div>
+  );
+}
+
+function SmallCircle(props) {
+  return (
+    <div>
+      <div style={{...mainStyle(props), ...styles.smallCircle}}>
+        <Content {...props} />
+      </div>
+    </div>
+  );
+}
+
+function LargeBubble(props) {
```

Comment:
Are these bubbles used in other places? I'm wondering if they should just be their own components

Reply:
currently they are not used in other places. however, we plan to refactor our existing `ProgressBubble` component (that this is based on) so that we don't have two versions. i haven't started looking at what all that will involve, but its easy to imagine that we might want to build it out of these subcomponents instead of building it on top of this whole component, at which point we'd want to move these out into a common module.